src/OrchardCore/OrchardCore.Abstractions/Json/Serialization/DateTimeJsonConverter.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace OrchardCore.Json.Serialization;
+public class DateTimeJsonConverter : JsonConverter<DateTime>
+{
+    // ISO 8601 format with milliseconds and time zone
+    private const string DateTimeFormat = "yyyy-MM-ddTHH:mm:ss.fffK"; 
+
+    public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType != JsonTokenType.String)
+        {
+            throw new JsonException($"Unexpected token parsing DateTime. Expected a string, got {reader.TokenType}.");
+        }
+
+        var stringValue = reader.GetString();
+        if (DateTime.TryParse(stringValue, null, DateTimeStyles.RoundtripKind, out var dateTime))
```

Comment:
There may be some discussion to be had here, such as intercultural compatibility issues

Reply:
Right

Reply:
@sebastienros do you have any feedback here?

Reply:
There is a sample implementation [here](https://learn.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support#-and-) that seem more optimized, and could use the `"o"` format automatically.